### ISSUE TITLE
console.log( sometimes becomes console(

### ISSUE THREAD

**Issue by EthanRutherford** [Microsoft/vscode#59848]
- VSCode Version: Code 1.27.2 (f46c4c469d6e6d8c46f268d1553c5dc4b475840f, 2018-09-12T16:17:45.060Z)
- OS Version: Windows_NT x64 10.0.17134

Steps to Reproduce:

1. type `console.log`
2. type `(`
3. (sometimes) `.log` disappears and I end up with `console(`, which is very clearly not what I wanted.

Does this issue occur when all extensions are disabled?: Yes

This is very confusing behavior, as it only happens *sometimes*, and I have not been able to figure out what conditions cause it to happen when it does. It's also confusing because I have no idea why anyone would even *want* this behavior, as `console()` is a TypeError unless there is some shadowing function named `console` somewhere.

**Comment by kylepaulsen** [comment]
I've noticed this in the past but haven't noticed it recently. I noticed it happened when my machine was running sorta slow and the autocomplete was still showing "console" when I typed ".log". Normally the autocomplete is supposed to immediately update to show what you can put after the dot when you type it, but because it was too slow, when you type the first parenthesis, the autocomplete will not realize you wanted something on the console object (because it failed to load the possible options in time!) and then you get console().

A (not really) workaround is to use this setting:
```
"editor.acceptSuggestionOnCommitCharacter": false
```

I was trying to look around in this file to see if I could notice any problems, but I couldn't figure anything out.... I even tried artificially slowing things down. https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/suggest/suggestModel.ts

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all extensions disabled?

**Comment by EthanRutherford** [comment]
Yes, it happens sporadically. Seems to happen more often in larger files, so I suspect @kylepaulsen's diagnosis is on the right track, and it is likely something to do with autocomplete not updating in time and acting on stale data.

**Comment by mjbvz** [comment]
Are you sure you have disabled extensions, any all TS server plugins? This sounds similar to https://github.com/Microsoft/typescript-styled-plugin/issues/73

**Comment by EthanRutherford** [comment]
No plugins, I'm not even using typescript.

**Comment by SirIntruder** [comment]
There is a very similar issue with c# as well
https://github.com/OmniSharp/omnisharp-vscode/issues/2472
